src/app/server/Dnssd.h

```diff
@@ -79,8 +80,11 @@ class DLL_EXPORT DnssdServer
 
     /// (Re-)starts the Dnssd server
     /// - if device has not yet been commissioned, then commissioning mode will show as enabled (CM=1, AC=0)
-    /// - if device has been commissioned, then commissioning mode will reflect the state of mode argument
-    void StartServer(chip::Dnssd::CommissioningMode mode = chip::Dnssd::CommissioningMode::kDisabled);
+    /// - if device has been commissioned, then commissioning mode will be disabled.
```

Comment:
I am curious why it's not the caller's job to determine whether the device is commissioned or not. Seems like a DNS-SD server should just advertise based on its configuration, rather than internally determine the configuration for some things and not others.

Reply:
I agree that we could have a totally different API shape for this server.  But I'm really trying to not boil this particular ocean right now, because I am not sure what that different shape would be.  _Something_ needs to coordinate the complicated "if this ifdef is set, advertise this way" dance we are doing, and right now it's the logic in `StartServer`.